### PR TITLE
Additional enemies, colors for enemy ships, and player ships selection

### PR DESCRIPTION
**Pull request by brewcoua**
This PR adds a few features:
- 3 additional enemy ships, with their own implementation
- Colors for each enemy ship
- A player ships selection menu in the settings (may be moved in the game settings later on)

Additionally, the rule excluding the `*.ttf` files in the `.gitignore` was removed as it ended up excluding the fonts file, which broke the entire game. Another thing is with the ship height offset being increased as it ended up outside of the screen on some laptops.
Lastly, this PR also cleans up and updates our requirements file with clearer roles assignments and requirements.

Here's a preview of the changes :
Screenshot from 2024-10-11 17-35-23

We are currently working on an additional enemy that will appear outside the defined enemy rectangle, having its own movement pattern.

### CODE REVIEW

**Review by bumsoft — APPROVED** [review]
LGTM:) However, I’m concerned that adding the font to the Git repository might raise copyright issues.

**Comment by brewcoua** [comment]
> LGTM:) However, I’m concerned that adding the font to the Git repository might raise copyright issues.

From the looks of it, the font does not have a specific licensing...
However, being considered "freeware", it should be fine being used in a non-commercial project, especially since it is mentionned in the README.md.

File: src/screen/GameScreen.java [hunk 1]
```diff
@@ -179,14 +179,14 @@ public final void initialize() {
...
-        this.ship = ShipFactory.create(this.shipType, this.width / 2, this.height - 30);
+        this.ship = ShipFactory.create(this.shipType, this.width / 2, this.height - 70);
         ship.applyItem(wallet);
 		//Create random Spider Web.
 		int web_count = 1 + level / 3;
 		web = new ArrayList<>();
 		for(int i = 0; i < web_count; i++) {
 			double randomValue = Math.random();
-			this.web.add(new Web((int) Math.max(0, randomValue * width - 12 * 2), this.height - 30));
+			this.web.add(new Web((int) Math.max(0, randomValue * width - 12 * 2), this.height - 70));
```
Comment (idealtrue0) [inline src/screen/GameScreen.java:193]: Is there a reason why the Y coordinates were changed? I'm curious if you have any!

Comment (brewcoua) [reply src/screen/GameScreen.java:193]: I faced an issue where the ship would simply be out of the screen, and only by increasing the offset would I be able to see it properly. I also increased it for the webs accordingly.

**Review by idealtrue0 — APPROVED** [review]
LGTM :)

**Review by fkrdud1125 — APPROVED** [review]
LGTM

File: src/screen/GameScreen.java (hunk 1 above)
Comment (dev-moonsc) [reply src/screen/GameScreen.java:193]: > I faced an issue where the ship would simply be out of the screen, and only by increasing the offset would I be able to see it properly. I also increased it for the webs accordingly.

Can you tell me exactly under what circumstances such an out-screen bug occurs?

Comment (brewcoua) [reply src/screen/GameScreen.java:193]: This is the preview before adding the offset:
[Screencast from 2024-10-12 21-59-51.webm](https://github.com/user-attachments/assets/57a221d3-b504-4a04-ac0e-2d6183afb457)

And now after adding it:
[Screencast from 2024-10-12 22-00-22.webm](https://github.com/user-attachments/assets/a99cee8b-45ed-4c2d-a31c-05e8d1f177af)

**Pull request merged**